reconcile/openshift_namespaces.py

```diff
@@ -1,127 +1,137 @@
 import logging
 
-import reconcile.utils.gql as gql
+from typing import List, Dict, Optional, Tuple, Any, Iterable, Mapping
 import reconcile.utils.threaded as threaded
-import reconcile.openshift_base as ob
 import reconcile.queries as queries
 
-from reconcile.utils.openshift_resource import ResourceInventory
 from reconcile.utils.oc import OC_Map
 from reconcile.utils.oc import StatusCodeError
 from reconcile.utils.defer import defer
 from reconcile.utils.sharding import is_in_shard
 
 
-QUERY = """
-{
-  namespaces: namespaces_v1 {
-    name
-    cluster {
-      name
-      serverUrl
-      jumpHost {
-          hostname
-          knownHosts
-          user
-          port
-          identity {
-              path
-              field
-              format
-          }
-      }
-      automationToken {
-        path
-        field
-        format
-      }
-      internal
-      disable {
-        integrations
-      }
-    }
-  }
-}
-"""
-
 QONTRACT_INTEGRATION = 'openshift-namespaces'
 
+NS_STATUS_PRESENT = "present"
+NS_STATUS_ABSENT = "absent"
+
+NS_ACTION_DELETE = "delete"
+NS_ACTION_CREATE = "create"
+
+
+def get_desired_state(
+        namespaces: Iterable[Mapping[str, Any]]) -> List[Dict[str, str]]:
 
-def get_desired_state(internal, use_jump_host, thread_pool_size):
-    gqlapi = gql.get_api()
-    all_namespaces = gqlapi.query(QUERY)['namespaces']
+    desired_state: List[Dict[str, str]] = []
+    for namespace in namespaces:
+        state = NS_STATUS_PRESENT
+        if namespace.get("delete"):
+            state = NS_STATUS_ABSENT
 
-    namespaces = []
+        desired_state.append({"cluster": namespace["cluster"]["name"],
+                              "namespace": namespace["name"],
+                              "state": state})
+
+    return desired_state
+
+
+def get_shard_namespaces() -> List[Dict[str, str]]:
+    all_namespaces = queries.get_namespaces(minimal=True)
+    namespaces = {}
+    to_be_ignored = set()
```

Comment:
Mmm... given the use you do below `duplicates` is probably a better name.

Reply:
Agree on this one. Changed.